### ISSUE TITLE
Adding prefix to metrics

### ISSUE THREAD

**Issue by daniyel** [jonnenauha/prometheus_varnish_exporter#9]
Hi.

It would be nice to add custom prefix to metrics. We are running Varnish as a component for microservices and we need to prefix the metrics with the name of microservice, so we can assign the metrics to the dashboard. I do not know if this is already possible with this version of exporter? 
Or maybe we could define our own namespace in `prometheus.go`?

Example:
`prometheus_varnish_exporter -web.prefix=my_service|-web.namespace=my_service:varnish_`

Output:
`my_service:varnish_main_busy_sleep 555`

Best regards.

**Comment by jonnenauha** [comment]
Is this `-web.namespace` common in other exporters and "supported" by prometheus itself?

**Comment by jonnenauha** [comment]
The feature itself is easy to add but I'd like to know if this is something Prometheus supports. Is there some spec for namespaces into metric names?

Maybe it would be better to do as a label? `-namespace xxx` > each having a `namespace` label like `varnish_main_busy_sleep{"namespace":xxx}`

Maybe `node` or `cluster` or something would be better?

**Comment by jonnenauha** [comment]
Relevant link https://prometheus.io/docs/practices/naming/

**Comment by daniyel** [comment]
Hm I do not know if there is any standard about this, but I just know from our experience because we have many services and each service can use for example `Varnish` with `prometheus_varnish_exporter`, then in Grafana under Metric lookup, I would get the same metrics for all services that use `prometheus_varnish_exporter` or I can seperate from which service the metrics come? 
For example i have two services. Both use `prometheus_varnish_exporter`. When they output metrics in our Grafana I would see same metrics for both? Therefore I cannot know which metric belongs to which varnish instance.

So with your solution the output for `varnish_lock_created{target="backend"}` would be?

**Comment by jonnenauha** [comment]
Yes, you could attach a unique label of your choosing for each `prom_varnish_exporter` instance when you start it.

But I believe this is a solved problem in Prometheus. In your prometheus servers config where you define the `prom_varnish_exporter` instances, you can attach labels to them. I have used this myself as I had two varnish instances and wanted one dashboard for each.

I assume you are using the new(er) Prometheus that has this new `static_config`  section https://prometheus.io/docs/operating/configuration/#<static_config>

I believe you can do it with this config [see example](https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus.yml)

```yml
scrape_configs:

  - job_name: 'prometheus dynamic stuff'

    static_configs:
      targets: 
        - 'first.varnish.exporter.com:9090'
      labels:
        cluster: dynamic_stuff

  - job_name: 'prometheus static stuff'

    static_configs:
      targets: 
        - 'second.varnish.exporter.com:9090'
      labels:
        cluster: static_stuff
      
```

I'm not exactly sure if the syntax is correct. The you can see my how to template your dashboard for "static_stuff" and "dynamic_stuff" with the "cluster" label https://github.com/jonnenauha/prometheus_varnish_exporter/issues/2#issuecomment-230533668

**Comment by daniyel** [comment]
But would I need to change the prometheus config each time I have new instance of Varnish? That would mean, each time some microservice uses varnish exporter I would need to add new scraper to the prometheus config?

**Comment by jonnenauha** [comment]
Don't you have to change the config each time anyway when a new varnish instance is set up? I mean even if this exporter would support those labels from cmd line, you would still need to add the exporter to your config.

Maybe you can open up your setup a bit more so I can understand. If you have a domain name for each type of service with changing IPs behind it that you manage with a DNS entry, you can just add those domain names to your prometheus config once (and setup the labels for them like my example).

**Comment by jonnenauha** [comment]
I implemented travis-ci build and test for all commits. At the same time I made it automatically upload my build scripts packages to github for a release. Proper releases are now a git tag away.

https://github.com/jonnenauha/prometheus_varnish_exporter/releases/tag/1.2

If you still find problems I can fix and make 1.2.1 release.

**Comment by daniyel** [comment]
Hi.

First sorry for not responding earlier. I think with label it would be also great, atleast it would follow the convention of prometheus. Sorry for the dumb question. Did you manage to do any changes with defining the label? I checked the readme and could not find andthing about defining the label for metrics.

Best regards.
